RHDEVDOCS-5246: Tekton Chains configuration changes for GA (#59627)
openshift/openshift-docs: modules/op-supported-keys-tekton-chains-configuration.adoc

|

| `builder.id`
| The builder ID to set for in-toto attestations
|
| `+https://tekton.dev/chains/v2+`

|===

If you enable the `docdb` storage method is for any artifacts, configure docstore storage options. For more information about the go-cloud docstore URI format, see the link:https://gocloud.dev/howto/docstore/[docstore package documentation]. {pipelines-title} supports the following docstore services:

* `firestore`
* `dynamodb`

.Chains configuration: Supported keys for docstore storage
[options="header"]
|===

| Key | Description | Supported values | Default value

| `storage.docdb.url`
| The go-cloud URI reference to a docstore collection. Used if the `docdb` storage method is enabled for any artifacts.
| `firestore://projects/[PROJECT]/databases/(default)/documents/[COLLECTION]?name_field=name`
|

|===

> If you enable the `grafeas` storage method for any artifacts, configure Grafeas storage options. For more information about Grafeas notes and occurrences, see link:https://github.com/grafeas/grafeas/blob/master/docs/grafeas_concepts.md[Grafeas concepts].

Author: I believe we are, as long as it's not upstream for our own products.

.Chains configuration: Supported keys for Grafeas storage
[options="header"]
|===

| Key | Description | Supported values | Default value

| `storage.grafeas.projectid`
| The {OCP} project in which the grafeas server for storing occurrences is located
|
|

| `storage.grafeas.noteid`
| Optional: the prefix to use for the name of all created notes.
| A string without spaces.
|

| `storage.grafeas.notehint`
| Optional: the https://github.com/grafeas/grafeas/blob/cd23d4dc1bef740d6d6d90d5007db5c9a2431c41/proto/v1/attestation.proto#L49[human_readable_name] field for the Grafeas `ATTESTATION` note.
|
|`This attestation note was generated by Tekton Chains`
|===

Optionally, you can enable additional uploads of binary transparency attestations.

.Chains configuration: Supported keys for transparency attestation storage
[options="header"]
|===

| Key | Description | Supported values | Default value

| `transparency.enabled`
| Enable or disable automatic binary transparency uploads.
| `true`, `false`, `manual`
| `false`

| `transparency.url`
| The URL for uploading binary transparency attestations, if enabled.
|
| `+https://rekor.sigstore.dev+`
|===

> NOTE: If you set `transparency.enabled` to `manual`, set the following annotation in `TaskRun` and `PipelineRun` resources to enable uploading them to the transparency log:

Reviewer: ```suggestion
NOTE: If you set `transparency.enabled` to `manual`, then only `TaskRuns` and `PipelineRuns` with the following annotation will be uploaded to the transparency log:
```

.Chains configuration: Supported keys for x509 keyless signing with Fulcio
[options="header"]
|===

| Key | Description | Supported values | Default value

| `signers.x509.fulcio.enabled`
| Enable or disable requesting automatic certificates from Fulcio.
| `true`, `false`
| `false`

| `signers.x509.fulcio.address`
| The Fulcio address for requesting certificates, if enabled
|
| `+https://v1.fulcio.sigstore.dev+`

| `signers.x509.fulcio.issuer`
| The expected OIDC issuer.
|
| `+https://oauth2.sigstore.dev/auth+`

| `signers.x509.fulcio.provider`
| The provider from which to request the ID Token
| `google`, `spiffe`, `github`, `filesystem`
| {pipelines-title} attempts to use every provider

| `signers.x509.identity.token.file`
| Path to the file containing the ID Token.
|
|

| `signers.x509.tuf.mirror.url`
| The URL for the TUF server. `$TUF_URL/root.json` must be present.
|
| `+https://sigstore-tuf-root.storage.googleapis.com+`
|===

If you configure the `kms` signature backend, set the KMS configuration, including OIDC and Spire, as necessary.

.Chains configuration: Supported keys for KMS signing
[options="header"]
|===

> | Key | Description | Supported values | Default value

Reviewer 2: Does it make sense to have two completely empty columns here?

Reviewer: So upstream the two columns are empty, not sure if we should remove or keep this two empty columns

Author: This is not about upstream - this is about consistency with all the other tables under this same heading, where the same columns are not empty. I would prefer to have more information about these keys but I don't have any.
